src/api/app/controllers/webui/repositories_controller.rb

```diff
@@ -241,14 +244,47 @@ def remove_flag
       @main_object.store
       format.html { redirect_to(action: :index, project: params[:project], package: params[:package]) }
       format.js do
-        switch_to_webui2
         render 'change_flag'
       end
     end
   end
 
+  # POST flag/change/:project(/:package)
+  # TODO: when removing bento, remove the extra 'change' from the route, for
+  # now we need to avoid the clash with create_flag
+  def change_flag
+    check_ajax
+    required_parameters :flag, :command
+    set_webui2_views
+    authorize @main_object, :update?
+
+    flag_type = params[:flag]
+    follow_change_flag_command(flag_type)
+
+    locals = { user_can_set_flags: true, project: @project, package: params[:package] }
+    locals[:architectures] = @main_object.architectures.reorder('name').distinct
+    locals[:flags] = Flag::SpecifiedFlags.new(@main_object, flag_type)
+    locals[:table_id] = 'flag_table_' + flag_type
+
+    render partial: 'shared/repositories_flag_table', locals: locals
+  end
+
   private
 
+  def follow_change_flag_command(flag_type)
+    architecture = Architecture.from_cache!(params[:architecture]) unless params[:architecture].nil?
+
+    if params[:command] == 'remove'
+      @main_object.flags.of_type(flag_type).where(repo: params[:repository], architecture: architecture).delete_all
+    elsif params[:command] == 'set-disable'
+      @main_object.flags.create(flag: flag_type, repo: params[:repository], architecture: architecture, status: 'disable')
+    elsif params[:command] == 'set-enable'
+      @main_object.flags.create(flag: flag_type, repo: params[:repository], architecture: architecture, status: 'enable')
```

Comment:
Or merge the two conditions into one. They are almost the same:wink:

Reply:
I did, not sure you'll like my solution

Reply:
Fine with me. Note that in Ruby ^ and $ are marking start and end of line. \A and \z are for beginning and end of a string.
Not that it would matter here.

Reply:
good to know